### ISSUE TITLE
NVDA does not report the selected status of each character in Microsoft 365

### ISSUE THREAD

**Issue by cary-rowen** [nvaccess/nvda#11891]
**Please pay attention to my recording presentation and log, Thanks**

### Steps to reproduce:
1. Enter "winword" in the windows + R run dialog to open the Word document.
2. Enter some text, such as: I am a student.
3. Press the Home button to the beginning of the line.
4. Hold down Shift and press Right repeatedly.

### Actual behavior:
NVDA only reports the selected status of some characters.

### Expected behavior:
NVDA should report the selected status of each character

#### NVDA installed/portable/running from source:
installed and portable.

#### NVDA version:
 alpha-21447, NVDA2020.3

#### Windows version:
Windows 10 20H2 (64-bit) build 19042.630

#### Name and version of other software in use when reproducing the issue:
Microsoft 365 MSO (16.0.13426.20270) 64 Bit

#### Other information about your system:
If the following option is selected, the issue cannot be reproduced: NVDA Options ➡ Advanced ➡ Use UI Automation to access Microsoft Word document controls.

The issue is inseparable from the synthesizer.

Problematic synthesizers include: Windows OneCore, eSpeak NG and Nuance Vocalizer expressive.

But, Microsoft Speech API version 4 and Microsoft Speech API version 5 Looks normal.

#### Does the issue still occur after restarting your computer?
Yes.

#### Have you tried any other versions of NVDA? If so, please report their behaviors.
no

#### If addons are disabled, is your problem still occuring?
Yes

#### Did you try to run the COM registry fixing tool in NVDA menu / tools?
Yes.

[nvda.log](https://github.com/nvaccess/nvda/files/5626374/nvda.log)
[audio](https://update.viyf.org/mc/Record.mp3)

**Comment by cary-rowen** [comment]
@leonardder  @josephsl @feerrenrut

**Comment by cary-rowen** [comment]
Does anyone have the same problem as me?

**Comment by dingpengyu** [comment]
cc @feerrenrut

**Comment by Adriani90** [comment]
I cannot reproduce this issue. on Which characters exactly is this not announced?

**Comment by cary-rowen** [comment]
Hi,
Windows OneCore and Espeak did not announce the selected status of every character.
It should be reflected in my recording

**Comment by cary-rowen** [comment]
This problem only exists when NVDA + Space is switched to focus mode.
@Adriani90 @jcsteh @feerrenrut @josephsl

**Comment by bhavyashah** [comment]
I can confirm this issue using nVDA 2020.4 Beta 4 and Word 365. Akin to cary-, I just reproduced this bug in focus mode but was unable to in browse mode. This issue is admittedly inconsistent but certainly not infrequent. I was unable to reproduce it in a new Word document with a single line of text; but managed to replicate it in an existing, 10-page, formatted Word document. I have experienced this issue on and off while doing my college classwork in Word as well. At those times, it does get vexing to have this hit-and-miss reporting of characters by NVDA when arrowing around rapidly.

**Comment by cary-rowen** [comment]
Thanks @bhavyashah,
Can anyone confirm this issue

**Comment by Adriani90** [comment]
@cary-rowen can you please test with NVDA last alpha version? Are you still having this issue when UIA for Microsoft Word is disabled?

**Comment by cary-rowen** [comment]
I can no longer reproduce the issue at the moment, but I just had the exact same issue reported by another user the other day.
The relevant information that cannot reproduce the problem is as follows
* alpha-28111
* Microsoft 365MSO (Version 2303 Build 16.0.16227.20202) 64 Bit
* Windows 10 22H2 (AMD64) build 19045.2846

I'm closing this issue for now, and will reopen it if more people report this issue.
